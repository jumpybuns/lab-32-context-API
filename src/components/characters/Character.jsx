import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ image, name }) => (
  <figure>
    <img src={image} alt={name} />
    <figcaption>{name}</figcaption>
  </figure>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default Character;
