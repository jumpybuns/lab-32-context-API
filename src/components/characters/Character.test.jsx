import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(
      <Character
        name="Morty Smith"
        image="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
