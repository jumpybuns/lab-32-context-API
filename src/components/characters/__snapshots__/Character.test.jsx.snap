// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Morty Smith"
      src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
    />
    <figcaption>
      Morty Smith
    </figcaption>
  </figure>
</DocumentFragment>
`;
